### PR TITLE
NetGsm Channel

### PR DESCRIPTION
**Pull request by zek**

### CODE REVIEW

**Comment by atymic** [comment]
Hi!

Are you still open to creating and maintaining this channel?

If so, could you update it to support laravel 5.5-6.0 please?

Thanks!

**Comment by zek** [comment]
I don't know if it's a problem but it only supports Laravel 6 right now.

**Review by atymic — DISMISSED** [review]

**Comment by atymic** [comment]
@zek 

Have you had a chance to look over those requested changes?

**Comment by zek** [comment]
@atymic Sorry I forgot this PR :) 

Can you check if everything is ok?

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{
+    "name": "laravel-notification-channels/netgsm",
+    "description": "NetGSM notification channel for Laravel 5.3 - 5.7",
```
Comment (atymic) [inline composer.json]:
```suggestion
    "description": "NetGSM notification channel for Laravel 5.5 & 6.x"
```

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            "name": "Talha Zekeriya Durmuş",
+            "email": "drtzack@gmail.com",
+            "homepage": "https://talhadurmus.com",
+            "role": "Developer"
+        }
+    ],
+    "require": {
+        "php": "^7.2",
+        "guzzlehttp/guzzle": "^6.3",
+        "illuminate/notifications": "5.5.*|5.6.*|5.7.*|5.8.*|^6.0",
```
Comment (atymic) [inline composer.json]:
```suggestion
        "illuminate/notifications": "~5.5 || ~6.0",
```

(and for support)

File: src/NetGsmChannel.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+     *
+     * @param  mixed  $notifiable
+     * @param  \Illuminate\Notifications\Notification  $notification
+     *
+     * @throws \NotificationChannels\NetGsm\Exceptions\CouldNotSendNotification
+     */
+    public function send($notifiable, Notification $notification)
+    {
+        $message = $notification->toNetGsm($notifiable);
+        if (is_string($message)) {
```
Comment (atymic) [inline src/NetGsmChannel.php]: could we have a little spacing between all the statements please

File: src/NetGsmClient.php [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+     */
+    protected $secret;
+
+    /**
+     * @var string
+     */
+    protected $msgHeader;
+
+    /**
+     * NetGsmClient constructor.
```
Comment (atymic) [inline src/NetGsmClient.php]: comment could be removed here :)

File: src/NetGsmClient.php [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    /**
+     * Send the Message.
+     * @param  NetGsmMessage  $message
+     * @throws CouldNotSendNotification
+     * @throws InvalidConfiguration
+     * @throws Exception
+     */
+    public function send(NetGsmMessage $message)
```
Comment (atymic) [inline src/NetGsmClient.php]: worth adding a return void typehint?

File: src/NetGsmClient.php [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+        $msg =
+            "<?xml version='1.0' encoding='utf-8'?>".
+            '<mainbody>'.
+            $this->prepareHeader($message).
+            $this->prepareBody($message).
+            '</mainbody>';
+
+        try {
+            $response = $this->client->request('POST', self::URI, [
```
Comment (atymic) [inline src/NetGsmClient.php]:
```suggestion
            $response = $this->client->post(self::URI, [
```

File: src/NetGsmClient.php [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+                throw InvalidConfiguration::invalidCredentials();
+            } elseif ($result[0] === '40') {
+                throw CouldNotSendNotification::invalidHeader();
+            } elseif ($result[0] === '70') {
+                throw CouldNotSendNotification::invalidRequest();
+            } else {
+                throw CouldNotSendNotification::unknownError();
+            }
+        } catch (InvalidConfiguration $exception) {
+            /// do nothing
```
Comment (atymic) [inline src/NetGsmClient.php]: Worth at least logging this? (call `report($exception)`)

File: src/NetGsmClient.php [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+        } catch (Exception $exception) {
+            throw CouldNotSendNotification::serviceRespondedWithAnError($exception);
+        }
+    }
+
+    /**
+     * @param  NetGsmMessage  $message
+     * @return string
+     */
+    protected function prepareBody(NetGsmMessage $message)
```
Comment (atymic) [inline src/NetGsmClient.php]: string return type

File: src/NetGsmClient.php [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+            '<msg><![CDATA['.$message->body.']]></msg>'.
+            $recipients.
+            '</body>';
+    }
+
+    /**
+     * @param  NetGsmMessage  $message
+     * @return string
+     */
+    protected function prepareHeader(NetGsmMessage $message)
```
Comment (atymic) [inline src/NetGsmClient.php]: string return type

File: src/NetGsmMessage.php [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace NotificationChannels\NetGsm;
+
+class NetGsmMessage
+{
+    public $body = '';
+    public $header = null;
+    public $recipients = [];
+
+    public static function create($body = '')
```
Comment (atymic) [inline src/NetGsmMessage.php]: could add self return type to all these methods :)

**Review by atymic — CHANGES_REQUESTED** [review]
Looks good!
Just a few minor things. Let me know once fixed and i'll create the repo for you :)

File: src/NetGsmClient.php (hunk 7 above)
Comment (zek) [reply src/NetGsmClient.php]: My mistake I don't know what I was thinking :) I changed it as re throw exception. Is that okay?

Comment (atymic) [reply src/NetGsmClient.php]: yep, that's fine :)

File: composer.json [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+            "email": "drtzack@gmail.com",
+            "homepage": "https://talhadurmus.com",
+            "role": "Developer"
+        }
+    ],
+    "require": {
+        "php": "^7.2",
+        "guzzlehttp/guzzle": "^6.3",
+        "illuminate/notifications": "~5.5 || ~6.0",
+        "illuminate/support": "5.5.*|5.6.*|5.7.*|5.8.*|^6.0"
```
Comment (atymic) [inline composer.json]: I'd make this the same as the `illuminate/notifications` dep
```suggestion
        "illuminate/support": "~5.5 || ~6.0"
```

**Review by atymic — APPROVED** [review]

**Comment by atymic** [comment]
I've create the repo:
https://github.com/laravel-notification-channels/netgsm
And launch checklist:
https://github.com/laravel-notification-channels/netgsm/issues/1

Please PR the final code there and i'll set all the services up :)

**Comment by atymic** [comment]
Closing this PR. Please respond on the other repo or i'll archive it shortly.

**Pull request closed without merging**